perses/annihilation/relative.py

```diff
@@ -2629,3 +2629,2206 @@ def _handle_exceptions(self):
+        whether to use a steric dispersion correction (not recommended for neq switching)
+    generate_htf_for_testing : bool
+        whether to generate the htf for testing
+
+    """
+
+    from openmmtools.constants import ONE_4PI_EPS0
+
+    _default_nonbonded_expression_list = [
+
+        "U_electrostatics + U_sterics;",
+
+        # Define electrostatics functional form
+        f"U_electrostatics = {ONE_4PI_EPS0} * chargeProd  * erfc(alpha * r)/ r_eff_electrostatics;",
+
+        # Define sterics functional form
+        "U_sterics = 4 * epsilon * x * (x - 1.0);"
+        "x = (sigma / r_eff_sterics)^6;"
+
+        # Define chargeProd (with REST scaling)
+        "chargeProd = (charge1 * p1_rest_scale) * (charge2 * p2_rest_scale);",
+        "p1_rest_scale = select(1 - is_both_solvent, is_rest1 * lambda_rest_scale + is_nonrest_solute1 + is_nonrest_solvent1 * lambda_rest_scale, 1);",
+        "p2_rest_scale = select(1 - is_both_solvent, is_rest2 * lambda_rest_scale + is_nonrest_solute2 + is_nonrest_solvent2 * lambda_rest_scale, 1);",
+        "is_both_solvent = is_nonrest_solvent1 * is_nonrest_solvent2;",
+
+        # Define charge1 and charge2 (with alchemical scaling)
+        "charge1 = (is_unique_old1 * old_charge_scalar1) + (is_unique_new1 * new_charge_scalar1) + is_core1 * (old_charge_scalar1 + new_charge_scalar1) + is_environment1 * (old_charge_scalar1 + new_charge_scalar1);",
+        "old_charge_scalar1 = lambda_electrostatics_old * charge_old1;",
+        "new_charge_scalar1 = lambda_electrostatics_new * charge_new1;",
+
+        "charge2 = (is_unique_old2 * old_charge_scalar2) + (is_unique_new2 * new_charge_scalar2) + is_core2 * (old_charge_scalar2 + new_charge_scalar2) + is_environment2 * (old_charge_scalar2 + new_charge_scalar2);",
+        "old_charge_scalar2 = lambda_electrostatics_old * charge_old2;",
+        "new_charge_scalar2 = lambda_electrostatics_new * charge_new2;",
+
+        # Define sigma
+        "sigma = (sigma1 + sigma2) / 2;",
+
+        # Define sigma1 and sigma2 (with alchemical scaling)
+        "sigma1 = (is_unique_old1 * old_sigma_scalar1) + (is_unique_new1 * new_sigma_scalar1) + is_core1 * (old_sigma_scalar1 + new_sigma_scalar1) + is_environment1 * (old_sigma_scalar1 + new_sigma_scalar1);",
+        "old_sigma_scalar1 = lambda_sterics_old * sigma_old1;",
+        "new_sigma_scalar1 = lambda_sterics_new * sigma_new1;",
+
+        "sigma2 = (is_unique_old2 * old_sigma_scalar2) + (is_unique_new2 * new_sigma_scalar2) + is_core2 * (old_sigma_scalar2 + new_sigma_scalar2) + is_environment2 * (old_sigma_scalar2 + new_sigma_scalar2);",
+        "old_sigma_scalar2 = lambda_sterics_old * sigma_old2;",
+        "new_sigma_scalar2 = lambda_sterics_new * sigma_new2;",
+
+        # Define epsilon (with rest scaling)
+        "epsilon = sqrt((epsilon1 * p1_rest_scale) * (epsilon2 * p2_rest_scale));",
+        # Q for DAR: Are we sure we want to apply the rest scaling before combining epsilons?
```

Comment:
no i'm not sure. it might make sense to combine eps with the mixing rule and then multiply by the scaling factor

Reply:
i think the maths require writing `(p1_rest_scale * p2_rest_scale * sqrt(epsilon1 * epsilon2))`

Reply:
Fixed